src/Nethermind/Nethermind.Synchronization/FastBlocks/SyncStatusList.cs

```diff
@@ -40,35 +39,15 @@ public void GetInfosForBatch(BlockInfo?[] blockInfos)
                     continue;
                 }
 
-                switch (_statuses[currentNumber])
+                if (_statuses.TrySet(currentNumber, FastBlockStatus.Sent, out FastBlockStatus status))
                 {
-                    case FastBlockStatus.Pending:
-                        if (!_cache.TryGet(currentNumber, out BlockInfo blockInfo))
-                        {
-                            blockInfo = _blockTree.FindCanonicalBlockInfo(currentNumber);
-                            if (blockInfo is not null)
-                            {
-                                _cache.Set(currentNumber, blockInfo);
-                            }
-                        }
-
-                        if (_statuses.TryMarkSent(currentNumber))
-                        {
-                            blockInfos[collected] = blockInfo;
-                            collected++;
-                        }
```

Comment:
If this returned false, we wouldn't go into `case FastBlockStatus.Inserted`, now it is simplified.